rs/replicated_state/src/canister_state/tests.rs

```diff
@@ -485,8 +485,8 @@ fn canister_state_induct_messages_to_self_guaranteed_response_duplicate_of_pause
     );
     // Nothing was enqueued.
     assert!(!fixture.canister_state.has_input());
-    // And the response is still in the output queue.
-    assert!(fixture.canister_state.has_output());
+    // And the response is gone from the output queue.
+    assert!(!fixture.canister_state.has_output());
```

Comment:
The response here is now expected to be gone because we handle the aborted/paused case as a silent drop as opposed to delegating to MR, right?

Reply:
Correct.

We used to handle the aborted/paused execution case explicitly by bailing out (because I suppose the check was added back when we only had guaranteed response messages, so a response for an already executing message was an indication of a bug). And despite what the test name says, what we're testing here are best-effort messages. For those we indeed silently drop duplicates of already executing responses.

Reply:
Because of the conflict between the test name ("guaranteed response") and the implementation (best effort), I ended up splitting the test into two: one for a guaranteed response duplicate (should be kept in place for Message Routing to deal with) and one for a best effort duplicate (should be silently consumed).